contract-examples/test/ExampleRewardManager.ts

```diff
@@ -0,0 +1,154 @@
+// (c) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import {
+  BigNumber,
+  Contract,
+  ContractFactory,
+} from "ethers"
+import { ethers } from "hardhat"
+import ts = require("typescript");
+
+// make sure this is always an admin for reward manager precompile
+const adminAddress: string = "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
+const REWARD_MANAGER_ADDRESS = "0x0200000000000000000000000000000000000004";
+const BLACKHOLE_ADDRESS = "0x0100000000000000000000000000000000000000";
+
+const ROLES = {
+  NONE: 0,
+  ENABLED: 1,
+  ADMIN: 2
+};
+
+describe("ExampleRewardManager", function () {
+  this.timeout("30s")
+  let owner: SignerWithAddress
+  let contract: Contract
+  let signer1: SignerWithAddress
+  let signer2: SignerWithAddress
+  let precompile: Contract
+
+  before(async function () {
+    owner = await ethers.getSigner(adminAddress);
+    signer1 = (await ethers.getSigners())[1]
+    signer2 = (await ethers.getSigners())[2]
+    const Contract: ContractFactory = await ethers.getContractFactory("ExampleRewardManager", { signer: owner })
+    contract = await Contract.deploy()
+    await contract.deployed()
+    const contractAddress: string = contract.address
+    console.log(`Contract deployed to: ${contractAddress}`)
+
+    precompile = await ethers.getContractAt("IRewardManager", REWARD_MANAGER_ADDRESS, owner);
+
+    // Send a transaction to mine a new block
+    const tx = await owner.sendTransaction({
+      to: signer1.address,
+      value: ethers.utils.parseEther("10")
+    })
+    await tx.wait()
+  });
+
+  it("should add contract deployer as owner", async function () {
```

Comment:
Can we add one final hardhat test for `disableRewards()`? Seems this is not tested in the HardHat tests atm.

Reply:
added